test/addons-abi/test_buffer/test_buffer.cc

```diff
@@ -0,0 +1,141 @@
+#include <string.h>
+#include <string>
+#include <node_api_helpers.h>
+
+#define JS_ASSERT(env, assertion, message) \
+  if (!(assertion)) { \
+    napi_throw_error((env), (std::string("assertion (" #assertion ") failed: ") + message).c_str()); \
```

Comment:
If the message is a string literal, then it can be concatenated at compile time, just like `#assertion`.

But I suppose requiring a string literal might not be ideal. And it's not like we care about performance here. So, feel free to disregard this suggestion.